crates/ruma-common/src/events/room/message.rs

```diff
@@ -299,6 +340,44 @@ impl RoomMessageEventContent {
     pub fn body(&self) -> &str {
         self.msgtype.body()
     }
+
+    /// Sanitize this message.
+    ///
+    /// If this message contains HTML, this removes the [tags and attributes] that are not listed in
+    /// the Matrix specification.
+    ///
+    /// It can also optionally remove the [rich reply fallback] from the plain text and HTML
+    /// message.
+    ///
+    /// This method is only effective on text, notice and emote messages.
+    ///
+    /// [tags and attributes]: https://spec.matrix.org/v1.2/client-server-api/#mroommessage-msgtypes
+    /// [rich reply fallback]: https://spec.matrix.org/v1.2/client-server-api/#fallbacks-for-rich-replies
+    #[cfg(feature = "sanitize")]
+    pub fn sanitize(
+        &mut self,
+        mode: HtmlSanitizerMode,
+        remove_reply_fallback: RemoveReplyFallback,
+    ) {
+        match &mut self.msgtype {
+            MessageType::Emote(EmoteMessageEventContent { body, formatted, .. })
+            | MessageType::Notice(NoticeMessageEventContent { body, formatted, .. })
+            | MessageType::Text(TextMessageEventContent { body, formatted, .. }) => {
+                formatted.as_mut().map(|formatted| {
+                    if let Some(body) = formatted.sanitize_html(mode, remove_reply_fallback) {
+                        formatted.body = body;
+                    }
+                    formatted
+                });
```

Comment:
This should be equivalent, right?

```suggestion
                if let Some(formatted) = formatted {
                    if let Some(body) = formatted.sanitize_html(mode, remove_reply_fallback) {
                        formatted.body = body;
                    }
                }
```